geode-core/src/main/java/org/apache/geode/internal/cache/entries/AbstractRegionEntry.java

```diff
@@ -744,7 +745,16 @@ public boolean initialImageInit(final InternalRegion region, final long lastModi
             }
           }
         }
-        setValue(region, this.prepareValueForCache(region, newValueToWrite, false));
+        Object value = null;
+        try {
+          value = this.prepareValueForCache(region, newValueToWrite, false);
+          setValue(region, value);
+        } catch (RegionDestroyedException e) {
+          if (value != null && value instanceof StoredObject) {
```

Comment:
I think the null check is not needed because null is never an "instanceof" anything.

Reply:
It is actually the job of the "setValue" method to release the offheap value it is given if it can not store it in the entry.
I see it calls "releaseOffHeapRefIfRegionBeingClosedOrDestroyed" so why didn't that work?
Did the RegionDestroyedException come from somewhere else?

Doing it the way you have it could lead to extra releases. I suggest you figure out why setValue did not release it since it was responsible for doing so and not to change the code at this location.

Reply:
The reguo destroyed exception is being thrown in _setValue.  This is called before the releaseoffheap... Method is called.  I can add the try catch to abstractregionentry instead.  The problem was the release here was not getting a chance to execute.  The stack trace is in the ticket.  

Ill move the try catch lower and see if that resolves the issue

Reply:
I don't think it is _setValue.
I think the bug is in the disk implementation of setValue:
at org.apache.geode.internal.cache.entries.DiskEntry$Helper.update(DiskEntry.java:856)

and also in basicUpdate that this method calls.
If either of these throw an exception before storing the new value in the entry then they are responsible for releasing the new value if it is a StoredObject.